iquip/apps/builder.py

```diff
@@ -365,22 +355,146 @@ def get_state(self, argInfo: Dict[str, Any]) -> Dict[str, Any]:
+          start: The start point for the RangeScan sequence.
+          stop: The end point for the RangeScan sequence.
+          npoints: The number of points in the RangeScan squence.
+          randomize: A boolean value that decides whether to shuffle the RangeScan sequence.
+        start: QDoubleSpinBox for start arguments inside state.
+        stop: QDoubleSpinBox for stop arguments inside state.
+        npoints: QSpinBox for npoints arguments inside state.
+        randomize: QCheckBox for randomize arguments inside state.
+        layout: The layout of _RangeScan widget.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ):
+        """Extended.
+
+        Args: See the docstring of Attributes for more detail.
+            procdesc: Each key and its value are as follows.
+              unit: The unit of the number value.
+              scale: See scale docstring at class Attributes.
+              global_step: The step between values changed by the up and down button.
+              global_min: The minimum value. (default=float("-inf"))
+              global_max: The maximum value. (default=float("inf"))
+                If min > max, then they are swapped.
+              ndecimals: The number of displayed decimals.
+            state: See the state docstring at class Attributes.
+        """
+        super().__init__(parent=parent)
+        self.scale = procdesc["scale"]
+        self.state = state
+        self.layout = QGridLayout(self)
+        self.start = QDoubleSpinBox(self)
+        self.apply_properties(self.start, procdesc)
+        self.start.setValue(state["start"] / self.scale)
+        self.npoints = QSpinBox(self)
+        self.npoints.setMinimum(1)
+        self.npoints.setMaximum((1 << 31) - 1)
+        self.npoints.setValue(state["npoints"])
+        self.stop = QDoubleSpinBox(self)
+        self.apply_properties(self.stop, procdesc)
+        self.stop.setValue(state["stop"] / self.scale)
+        self.randomize = QCheckBox("Randomize", self)
+        self.randomize.setChecked(state["randomize"])
+        #layout
+        self.layout.addWidget(QLabel("start:", self), 0, 0)
+        self.layout.addWidget(self.start, 0, 1)
+        self.layout.addWidget(QLabel("npoints:", self), 1, 0)
+        self.layout.addWidget(self.npoints, 1, 1)
+        self.layout.addWidget(QLabel("stop:", self), 2, 0)
+        self.layout.addWidget(self.stop, 2, 1)
+        self.layout.addWidget(self.randomize, 3, 1)
+        self.start.valueChanged.connect(self.update_start)
+        self.npoints.valueChanged.connect(self.update_npoints)
+        self.stop.valueChanged.connect(self.update_stop)
+        self.randomize.stateChanged.connect(self.update_randomize)
+
+    def apply_properties(self, widget: QDoubleSpinBox, procdesc: Dict[str, Any]):
+        """Adds properties to the SpinBox widget.
```

Comment:
If you want to write upper letters, it should be a valid class or variable name.
In other words, you should write "the spin box widget" or "the QDoubleSpinBox widget".